seed/seedwriter/writer_test.go

```diff
@@ -3668,25 +3698,41 @@ func (s *writerSuite) TestSeedSnapsWriteMetaCore20ExtraSnaps(c *C) {
 	snapDeclUniq := func(snapName string) string {
 		return "snap-declaration/16/" + s.AssertedSnapID(snapName)
 	}
+	snapResPairUniq := func(snapName, compName string, resRev, snapRev int) string {
+		return fmt.Sprintf("snap-resource-pair/%s/%s/%d/%d", s.AssertedSnapID(snapName), compName, resRev, snapRev)
+	}
 
-	c.Check(seen, DeepEquals, map[string]bool{
-		"account/developerid":                                             true,
-		snapDeclUniq("core18"):                                            true,
-		snapDeclUniq("cont-producer"):                                     true,
-		snapDeclUniq("required20"):                                        true,
-		snapRevUniq("core18", 1):                                          true,
-		snapRevUniq("cont-producer", 1):                                   true,
-		snapRevUniq("required20", 21):                                     true,
-		resRevUniq("required20", "comp1", 22):                             true,
-		resRevUniq("required20", "comp2", 33):                             true,
-		"snap-resource-pair/required20ididididididididididid/comp1/22/21": true,
-		"snap-resource-pair/required20ididididididididididid/comp2/33/21": true,
+	c.Check(seenExtra, DeepEquals, map[string]bool{
+		"account/developerid":                          true,
+		snapDeclUniq("core18"):                         true,
+		snapDeclUniq("cont-producer"):                  true,
+		snapDeclUniq("required20"):                     true,
+		snapRevUniq("core18", 1):                       true,
+		snapRevUniq("cont-producer", 1):                true,
+		snapRevUniq("required20", 21):                  true,
+		resRevUniq("required20", "comp1", 22):          true,
+		resRevUniq("required20", "comp2", 33):          true,
+		snapResPairUniq("required20", "comp1", 22, 21): true,
+		snapResPairUniq("required20", "comp2", 33, 21): true,
 	})
 
+	// all of these end up in the main assertions file since the snap is
+	// asserted and in the model, despite the component not being in the model.
+	// this might be something we want to change
```

Comment:
how painful it is to be consistent here?

Reply:
Quite annoying from my first attempt but I can make another attempt today.

Reply:
I'm going to open a follow-up PR that removes the extra-snaps file.